openmrs/omod/src/main/java/org/projectbuendia/openmrs/web/controller/ProfileManager.java

```diff
@@ -145,68 +175,79 @@ String getNextVersionedFilename(String name) {
     }
 
     /** Handles an uploaded profile. */
-    // TODO: Give some error message if the profile is not uploaded correctly
-    //       (When tomcat7 didn't have permission to PROFILES_DIR it was failing
-    //        without showing any error message to the user)
-    void addProfile(MultipartHttpServletRequest request, ModelMap model) {
+    private void addProfile(MultipartHttpServletRequest request, ModelMap model) {
         List<String> lines = new ArrayList<>();
         MultipartFile mpf = request.getFile("file");
         if (mpf != null) {
+            String message = "";
+            String filename = mpf.getOriginalFilename();
             try {
                 File tempFile = File.createTempFile("profile", null);
                 mpf.transferTo(tempFile);
-                if (execute(VALIDATE_CMD, tempFile, lines)) {
-                    String filename = getNextVersionedFilename(mpf.getOriginalFilename());
-                    File newFile = new File(PROFILE_DIR, filename);
-                    model.addAttribute("filename", filename);
+                boolean exResult = execute(VALIDATE_CMD, tempFile, lines);
```

Comment:
Okay.  I guess I'll provide my reasoning for why I think it's not a good way to spend a line of code.

1. It makes the code harder to read because you have to say in your mind "We are going to `execute(...)` and store the success flag in `exResult`, and then we are going to test `exResult`, and if it is true then..." instead of "If `execute(...)` succeeds then..."  That's a lot more thinking.

2. It makes the code harder to change because if you edit the `execute(...)` call, you don't know whether, by changing the meaning of `exResult`, you have affected the behaviour of the program many lines down below at a later point where it happens to use `exResult` again.  (Or, in other words, the variable `exResult` has excessively large scope.)  If you write the expression without `exResult`, it is obvious that editing the `execute(...)` call affects this one `if` statement and cannot affect anything else.

3. Being able to tell whether a line of code has a side effect is important, I agree.  But that is not the responsibility of the `if` statement; it is the responsibility of the method name, `execute`, which is pretty clear that it has a side effect.

I don't think it's a really big deal in this case, but I did want to at least share my thinking with you so you understand where I'm coming from and have a chance to contemplate the benefits of the other way as well.